scarb/src/ops/lint.rs

```diff
@@ -0,0 +1,246 @@
+};
+use scarb_ui::components::Status;
+use smol_str::SmolStr;
+use std::cmp::Reverse;
+use std::collections::HashMap;
+
+use crate::core::{Package, Workspace};
+
+use super::{CompilationUnitsOpts, FeaturesOpts, FeaturesSelector};
+
+pub struct LintOptions {
+    pub packages: Vec<Package>,
+    pub test: bool,
+    pub fix: bool,
+}
+
+#[tracing::instrument(skip_all, level = "debug")]
+pub fn lint(opts: LintOptions, ws: &Workspace<'_>) -> Result<()> {
+    let feature_opts = FeaturesOpts {
+        features: FeaturesSelector::AllFeatures,
+        no_default_features: true,
+    };
+
+    let resolve = ops::resolve_workspace(ws)?;
+
+    let compilation_units = ops::generate_compilation_units(
+        &resolve,
+        &feature_opts,
+        ws,
+        CompilationUnitsOpts {
+            ignore_cairo_version: false,
+            load_prebuilt_macros: false,
+        },
+    )?;
+
+    for package in opts.packages {
+        let package_compilation_units = if opts.test {
+            compilation_units
+                .iter()
+                // TODO: Test this (if the test packages are also checked)
+                .filter(|compilation_unit| {
+                    let is_main_component = compilation_unit.main_package_id() == package.id;
+                    let has_test_components =
+                        compilation_unit.components().iter().any(|component| {
+                            component.target_kind() == TargetKind::TEST
+                                && component.package.id == package.id
+                        });
+                    let is_integration_test_package =
+                        compilation_unit.main_package_id().name.to_string()
+                            == format!("{}_integrationtest", package.id.name)
+                            && compilation_unit.main_package_id().version == package.id.version;
+                    is_main_component || has_test_components || is_integration_test_package
+                })
+                .collect::<Vec<_>>()
+        } else {
+            vec![compilation_units
+                .iter()
+                .find(|compilation_unit| compilation_unit.main_package_id() == package.id)
+                .unwrap()]
+        };
+
+        for compilation_unit in package_compilation_units {
+            match compilation_unit {
+                // We skip proc macros as we don't want to check anything related to rust code.
+                CompilationUnit::ProcMacro(_) => ws
+                    .config()
+                    .ui()
+                    .print(Status::new("Skipping proc macro", &compilation_unit.name())),
+                CompilationUnit::Cairo(compilation_unit) => {
+                    ws.config()
+                        .ui()
+                        .print(Status::new("Checking", &compilation_unit.name()));
+
+                    let additional_plugins = if opts.test {
+                        vec![
+                            test_plugin_suite(),
```

Comment:
This plugin conflicts with Starknet Foundry one. It is also explicitly declared by the user in deps. I think it should not be injected here.